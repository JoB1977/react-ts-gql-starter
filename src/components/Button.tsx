import { ButtonHTMLAttributes, DetailedHTMLProps, FC } from 'react';

const Button: FC<DetailedHTMLProps<ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>> = ({
  className,
  children,
  ...props
}) => (
  <button className={'bg-gray-300 px-4 rounded ' + className} {...props}>
    {children}
  </button>
);

export default Button;
