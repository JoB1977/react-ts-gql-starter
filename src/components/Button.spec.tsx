import { render } from '@testing-library/react';
import Button from './Button';

describe('Button', () => {
  it('Button should be created and match snapshot', () => {
    const fixture = render(<Button name="crazyButton">Click me</Button>);

    expect(fixture.container).toMatchSnapshot();
  });
});
