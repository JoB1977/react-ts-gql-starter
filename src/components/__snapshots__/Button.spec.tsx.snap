// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button should be created 1`] = `
<div>
  <button
    class="bg-gray-300 px-4 rounded "
    name="crazyButton"
  >
    Click me
  </button>
</div>
`;
